lib/src/embedded/utils.dart

```diff
@@ -68,3 +76,24 @@ T withGlyphs<T>(T callback(), {required bool ascii}) {
   term_glyph.ascii = currentConfig;
   return result;
 }
+
+/// Serializes [value] to an unsigned varint.
+Uint8List serializeVarint(int value) {
+  if (value == 0) return Uint8List.fromList([0]);
+  RangeError.checkNotNegative(value);
+
+  // `log2(wireId) = ln(wireId) / ln(2)` is the length in bits of the wire ID.
+  // Varints encode 7 bits to a byte, so we divide by 7 again and take the
+  // ceiling to find the total number of bytes we'll need to encode the full
+  // value. Then `(ln(wireId) / ln(2)) / 7 = ln(wireId) / (ln(2) * 7)` to
+  // ensure Dart constant-folds as much as possible.
+  var lengthInBytes = (math.log(value) / (math.ln2 * 7)).ceil();
```

Comment:
Dart has a `.bitLength` method. A simpler and more performant way to write this would be:

``` dart
  var lengthInBytes = (value.bitLength + 6) ~/ 7;
```

or

``` dart
  var lengthInBytes = (value.bitLength / 7).ceil(); 
```